lib/internal/child_process.js

```diff
@@ -699,6 +702,21 @@ function setupChannel(target, channel, serializationMode) {
     });
   });
 
+  target.on('newListener', function() {
+    if (!target.channel) return;
+
+    const messages = target.channel[kPendingMessages];
+
+    process.nextTick(() => {
+      if (!messages.size) return;
```

Comment:
Right now, this code queues up messages whenever there is no listener for messages, regardless of whether there has been one before. That's fine, but in that case, we should probably fetch the pending messages here in the nextTick callback, and, more importantly, check whether there still are listeners (because they could have been removed before nextTick fires)

Reply:
Amazing! I assumed that as I was handling the 'newListener'  event it must be someone attached. However, I didn't think about when the nextTick callback fires it might not have been someone attached. 

I fixed it on the new commit!